airflow/api_fastapi/core_api/serializers/connections.py

```diff
@@ -34,8 +36,23 @@ class ConnectionResponse(BaseModel):
     login: str | None
     schema_: str | None = Field(alias="schema")
     port: int | None
+    password: str | None
     extra: str | None
 
+    @field_validator("password", mode="after")
+    @classmethod
+    def redact_password(cls, v: str | None, field_info: ValidationInfo) -> str | None:
+        if v is None:
+            return None
+        try:
+            redacted_dict = json.loads(
+                json.dumps(redact(value={field_info.field_name: v}, name=field_info.field_name))
+            )
+            redacted_password = redacted_dict.get(field_info.field_name, "***")
+            return redacted_password
+        except (json.JSONDecodeError, ValueError) as ex:
+            raise ex
```

Comment:
I think all of that was necessary because the `val` of a Variable is a dictionary with some values to redact.

In this case we just need to redact the value, why can't we just `redact(self.val, "password")` ?

Reply:
This worked smoothly. Indeed, my mind just didn't accept the fact the same approach wouldn't work with the same structure. I totally missed this part where we can pass the name for a pre-defined hidden key like a password. :facepalm: Thanks! :) Rebased, passed the `field_name` ("password") to `redact()`